packages/backend-data/src/convert_schema.ts

```diff
@@ -61,3 +80,54 @@ export const convertSchemaToCDK = (
     provisionStrategy,
   });
 };
+
+/**
+ * Given an input list of CDK Graphql Def interface schemas, produce the relevant CDK Graphql Def interface
+ * @param schemas the cdk graphql definition interfaces to combine
+ * @returns the cdk graphql definition interface
+ */
+export const combineCDKSchemas = (
+  schemas: IAmplifyDataDefinition[]
+): IAmplifyDataDefinition => {
+  return AmplifyDataDefinition.combine(schemas);
+};
+
+/**
+ * Given the generated rds builder database configuration, convert it into the DataSource strategy
+ * @param configuration the database configuration from `data-schema`
+ * @returns the data strategy needed to configure the data source
+ */
+const convertDatabaseConfigurationToDataSourceStrategy = (
+  configuration: DataSourceConfiguration
+): ModelDataSourceStrategy => {
+  // DO NOT EDIT THE FOLLOWING VALUES, UPDATES TO DB TYPE OR STRATEGY WILL RESULT IN DB REPROVISIONING
+  const defaultDbType = 'DYNAMODB';
+  const defaultProvisionStrategy = 'AMPLIFY_TABLE';
+
+  const dbEngine = configuration.engine;
+
+  if (dbEngine === 'dynamodb') {
+    return {
+      dbType: defaultDbType,
+      provisionStrategy: defaultProvisionStrategy,
+    };
+  }
+
+  const dbType = <Uppercase<typeof configuration.engine>>dbEngine.toUpperCase();
+
+  return {
+    dbType,
+    name: '',
+    dbConnectionConfig: {
+      // These are all about to be replaced
+      databaseNameSsmPath: '',
+      hostnameSsmPath: '',
+      passwordSsmPath: '',
+      portSsmPath: '',
+      usernameSsmPath: '',
+      // Replacement mapping
+      // connectionUriSsmPath: configuration.connectionUri
+    },
+    vpcConfiguration: configuration.vpcConfig,
```

Comment:
I changed the front end types to exactly conform with the backend. I don't like that these types are redefined separately.  It's unnecessarily fragile.

Reply:
Can you explain this a bit more? Where are the two definitions located?

Reply:
The configured vpc params defined by [this shape](https://github.com/stocaaro/amplify-api-next/blob/plumbing/packages/data-schema-types/src/builder/types.ts#L79) are eventually consumed under [this shape](https://github.com/aws-amplify/amplify-category-api/blob/b4de9a2e8c62f124babd2a25557cc1bde8b0b896/packages/amplify-graphql-api-construct/src/model-datasource-strategy-types.ts#L87)

My concern may be unwarranted, it just seems like defining the shape in 2 places and depending upon homomorphism to tell is if we've broken the expectation is less clean than sharing a single base type.